iohub/ngff_utils.py

```diff
@@ -0,0 +1,341 @@
+import contextlib
+import inspect
+import io
+import itertools
+import multiprocessing as mp
+from functools import partial
+from pathlib import Path
+from typing import Tuple
+
+import click
+import numpy as np
+from numpy.typing import DTypeLike
+
+from iohub.ngff import Position, open_ome_zarr
+from iohub.ngff_meta import TransformationMeta
+
+
+def create_empty_hcs_zarr(
+    store_path: Path,
+    position_keys: list[Tuple[str]],
+    channel_names: list[str],
+    shape: Tuple[int],
+    chunks: Tuple[int] = None,
+    scale: Tuple[float] = (1, 1, 1, 1, 1),
+    dtype: DTypeLike = np.float32,
+    max_chunk_size_bytes=500e6,
+) -> None:
+    """
+    This function creates a new HCS Plate in OME-Zarr format if the plate does not exist.
+
+    If the plate exists, appends positions and channels if they are not
+    already in the plate.
+
+    Parameters
+    ----------
+    store_path : Path
+        hcs plate path
+    position_keys : list[Tuple[str]]
+        Position keys, will append if not present in the plate.
+        e.g. [("A", "1", "0"), ("A", "1", "1")]
+    shape : Tuple[int]
+    chunks : Tuple[int]
+    scale : Tuple[float]
+    channel_names : list[str]
+        Channel names, will append if not present in metadata.
+    dtype : DTypeLike
+    """
```

Comment:
Add documentation on all parameters, particularly how we handle chunk sizes. What would happen if the there is already a zarr store at `store_path` with FOVs of different shape / chunks / dtype? A store like that is valid and `iohub` would allow you to save data like that, but viewers may be unhappy. `iohub info` reports a chunk size - what would happen if FOVs use different chunking? Should we check for that and disallow it in this function?

Reply:
Added documentation to all the parameters. I guess that if the the store already exists and `chunks` is not the same as the current one, then we can raise a warning. If you create one from scratch I think `chunks` should be exposed.